controllers/metal3.io/host_state_machine.go

```diff
@@ -348,6 +348,8 @@ func (hsm *hostStateMachine) handleRegistering(info *reconcileInfo) actionResult
 	// if the credentials change and the Host must be re-registered.
 	if hsm.Host.Spec.ExternallyProvisioned {
 		hsm.NextState = metal3v1alpha1.StateExternallyProvisioned
+	} else if inspectionDisabled(hsm.Host) {
+		hsm.NextState = metal3v1alpha1.StatePreparing
```

Comment:
Need to also handle the transition from the ExternallyProvisioned state.

Reply:
I'm a bit blurry after the summit, could you elaborate? This is a branch for `!ExternallyProvisioned`

Reply:
There is another transition to Preparing on line 386, but I can't leave a comment there because GitHub, so I left it here.